### ISSUE TITLE
ArgumentOutOfRangeException when I try to edit a recipe

### ISSUE THREAD

**Issue by JimH44** [franc6/Ketarin#54]
In Ketarin-Mono version 1.0.41, when I select a recipe and tap Enter to edit it, this exception happens. This may have happened because of an automatic update of the Mono system. I'll look into that. Jim

See the error dump file 
[Edit error.txt](https://github.com/franc6/Ketarin/files/5186557/Edit.error.txt)

**Comment by JimH44** [comment]
Here's the current version of Mono:
jim@langtran:/home/jim$ mono --version
Mono JIT compiler version 6.12.0.90 (tarball Fri Sep  4 14:02:28 UTC 2020)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           __thread
	SIGSEGV:       altstack
	Notifications: epoll
	Architecture:  x86
	Disabled:      none
	Misc:          softdebug 
	Interpreter:   yes
	LLVM:          yes(610)
	Suspend:       hybrid
	GC:            sgen (concurrent by default)
========================
But that's strange. At https://www.mono-project.com , when I click Download I find myself at
https://www.mono-project.com/download/stable/ and there they say
"The latest Stable Mono release is: 6.10.0 Stable (6.10.0.104)"

**Comment by JimH44** [comment]
And my current version of Linux:
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.5 LTS
Release:	18.04
Codename:	bionic

**Comment by JimH44** [comment]
Well, this is strange. I still have Ketarin-Mono version 1.0.39 on my system and tried it, and with this one I can select a recipe and tap Enter to edit it and the edit dialog opens and works properly.

**Comment by franc6** [comment]
So far, I'm unable to reproduce this.  I did see another issue, which makes me wonder how build 41 even exists, as it seems it should have failed to build. :(

Before updating my system, I was running Mono version 6.10.0.104, which matches what the website indicates is the latest stable version, but running apt dist-upgrade installed 6.12.0.90, which is listed as a preview release, still.  I'm guessing somebody goofed, and accidentally published it to the stable repository.  Or perhaps somebody forgot to update the page?

Either way, I think I see why it might be crashing for you and not for me.  If your database was corrupted in some way (or possibly a mismatch between SQLite versions it was compiled for versus what it's running with*), it might be reading a value too high for the "total number of files to keep (original + backups)", which would cause the error you are seeing.  I'll add some code to check the value it reads, and reset it to the default if it reads something out of range.  If you typically set that value to something else, you'll want to double-check them all in the next build. :(

This change will at least fix that particular crash, although, I obviously can't guarantee you won't see some other problem.

* That comment goes back to my earlier note about the problem I saw preventing builds -- it should not have compiled successfully, as I missed a typo when updating SQLite versions.  But I thought I had tested it as building properly before pushing to github, so I'm not sure what's going on.
